src/NorthwindWebsite/Middleware/ImageCachingMiddleware.cs

```diff
@@ -0,0 +1,122 @@
+﻿using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Constants;
+using NorthwindWebsite.Middleware.Handlers.Interfaces;
+using NorthwindWebsite.Presentation.Utils;
+
+namespace NorthwindWebsite.Middleware;
+
+public class ImageCachingMiddleware : IMiddleware
+{
+    private readonly AppSettings _appSettings;
+    private readonly IImageCachingHandler _imageCachingHandler;
+    private readonly ILogger<ImageCachingMiddleware> _logger;
+
+    public ImageCachingMiddleware(
+        AppSettings appSettings,
+        IImageCachingHandler cacheService,
+        ILogger<ImageCachingMiddleware> logger)
+    {
+        _appSettings = appSettings;
+        _imageCachingHandler = cacheService;
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        var isExpiresValueParsed = DateTime.TryParse(context.Response.Headers["expires"], out var expires);
+
+        var requestPath = context.Request.Path.Value;
+
+        var imageIndex = UrlUtils.GetImageIndexFromRequest(requestPath);
+
+        ValidateCachingDerectoryExistence();
+
+        if (isExpiresValueParsed && expires.Subtract(DateTime.Now) <= TimeSpan.Zero)
```

Comment:
Well, my first point of view was to throw an exception (or simply return bad request on invalid `expires` value)